components/arm.py

```diff
@@ -172,10 +172,14 @@ def update_display(self) -> None:
 
     def execute(self) -> None:
         self.update_display()
+        if self.is_retracted():
+            self.set_at_min_extension()
```

Comment:
I think we may need to call reset on the extension controller after changing it position